docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,422 @@
+- `ResizeMonitor` monitors the <visualize> for size change
+- on new data from `responseHandler` or on size change your visualization render method gets called.
+It should return a promise and resolve once its done rendering.
+- <visualize> will emit `renderComplete` event on the element once your render methods promise is resolved.
+- visualization should call `updateState()` any time something changed that requires to re-render or fetch new data.
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
+
+class MyVisualization {
+   constructor(vis, el) {
+      this.el = el;
+      this.vis = vis;
+   }
+   render(visData, status) {
+      return new Promise(resolve => {
+         ...
+         resolve('done rendering');
+      }),
+   destroy() {
+      console.log('destroying');
+   }
+}
+
+const MyNewVisType = (Private) => {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createBaseVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    visualization: MyVisualization
+  });
+}
+
+VisTypesRegistryProvider.register(MyNewVisType);
+-----------
+
+[[development-angular-visualization-type]]
+==== AngularJS Visualization Type
+AngularJS visualization type assumes you are using angular as your rendering technology. Instead of providing the
+controller we need to provide the angular template to render. 
+
+Visualization will receive vis, uiState and visData on the $scope and needs to call $scope.renderComplete() once 
+its done rendering.
+
+["source","js"]
+-----------
+const MyNewVisType = (Private) => {
+  const VisFactory = Private(VisFactoryProvider);
+  
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    visConfig: {
+       template: '<div ng-controller="MyAngularController"></div>`
+    }
+  });
+}
+-----------
+
+[[development-react-visualization-type]]
+==== React Visualization Type
+React visualization type assumes you are using React as your rendering technology. Instead of passing it an AngularJS
+template you need to pass React component.
+
+Visualization will receive vis, uiState and visData as props as well as the renderComplete which should be called once
+done rendering.
+
+["source","js"]
+-----------
+import { ReactComponent } from './my_react_component';
+
+const MyNewVisType = (Private) => {
+  const VisFactory = Private(VisFactoryProvider);
+  
+  return VisFactory.createReactVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    visConfig: {
+       template: ReactComponent
+    }
+  });
+}
+-----------
+
+[[development-vislib-visualization-type]]
+==== Vislib Visualization Type
```

Comment:
I would remove Vislib Visualization type imho. This is very much internal API, and would require a continuous effort to keep documentation in sync.

Reply:
it doesn't contain much info, just for the reasons you mentioned, but i think its worth mentioning it here, instead of pretending it does not exist.